async-query/src/main/java/org/opensearch/sql/spark/scheduler/model/OpenSearchRefreshIndexJobRequest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.scheduler.model;
+
+import java.io.IOException;
+import java.time.Instant;
+import lombok.Builder;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.schedule.Schedule;
+
+/** Represents a job request to refresh index. */
+@Builder
+public class OpenSearchRefreshIndexJobRequest implements ScheduledJobParameter {
+  // Constant fields for JSON serialization
+  public static final String JOB_NAME_FIELD = "jobName";
+  public static final String JOB_TYPE_FIELD = "jobType";
+  public static final String LAST_UPDATE_TIME_FIELD = "lastUpdateTime";
+  public static final String LAST_UPDATE_TIME_FIELD_READABLE = "last_update_time_field";
+  public static final String SCHEDULE_FIELD = "schedule";
+  public static final String ENABLED_TIME_FIELD = "enabledTime";
+  public static final String ENABLED_TIME_FIELD_READABLE = "enabled_time_field";
+  public static final String LOCK_DURATION_SECONDS = "lockDurationSeconds";
+  public static final String JITTER = "jitter";
+  public static final String ENABLED_FIELD = "enabled";
+
+  // name is doc id
+  private final String jobName;
+  private final String jobType;
+  private final Schedule schedule;
+  private final boolean enabled;
+  private final Instant lastUpdateTime;
+  private final Instant enabledTime;
+  private final Long lockDurationSeconds;
+  private final Double jitter;
+
+  @Override
+  public String getName() {
+    return jobName;
+  }
+
+  public String getJobType() {
+    return jobType;
+  }
+
+  @Override
+  public Schedule getSchedule() {
+    return schedule;
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  @Override
+  public Instant getLastUpdateTime() {
+    return lastUpdateTime;
+  }
+
+  @Override
+  public Instant getEnabledTime() {
+    return enabledTime;
+  }
+
+  @Override
+  public Long getLockDurationSeconds() {
+    return lockDurationSeconds;
+  }
+
+  @Override
+  public Double getJitter() {
+    return jitter;
+  }
```

Comment:
Can we use `@Data` or `@Getter` annotation on this class?

Reply:
`@Data` and `@Getter` should be able to implement methods from interface. Did you try?

Reply:
![Screenshot 2024-07-24 at 11 46 02](https://github.com/user-attachments/assets/ae943f6a-2e49-410e-918a-bb7e0bcda385)
after removed 
```
  @Override
  public String getName() {
    return jobName;
  }
```